### ISSUE TITLE
CLOSED:  Attributeerror:'NoneType' object has no attribute 'ids' #kivy # python

### ISSUE THREAD

**Issue by cuneyttanrisever** [kivy/kivy#7862]
While the code works without any error in **windows10**, **ubuntu** also gives this error, why?
how do i fix it?
Attributeerror:'NoneType' object has no attribute 'ids' 
py code:

```
# -*- coding: utf-8 -*-

from platform import machine, version, platform, uname, system, processor
from kivy.app import App
makineadi1=machine()
makine_versionu=version()
makine_platform=platform()
makine_adi=uname()
makine_system=system()
makine_islemci=processor()
class olayUyg(App):
    def build(self):
        self.root.ids.mesaj1.text='makine info: {} !'.format(makineadi1)
        self.root.ids.mesaj2.text='makine versionu: {} !'.format(makine_versionu)
        self.root.ids.mesaj3.text='makine platform: {} !'.format(makine_platform)
        self.root.ids.mesaj4.text='makine adı: {} !'.format(makine_adi)
        self.root.ids.mesaj5.text='makine system: {} !'.format(makine_system)
        self.root.ids.mesaj6.text='makine işlemci: {} !'.format(makine_islemci)
    def sifredogrula(self):
        kulladi1 = self.root.ids.kulladi.text
        kullsifre1=self.root.ids.kullsifre.text
        if kulladi1=="DexmoD" and kullsifre1=="DexmoD":
            self.root.ids.mesaj.text='Merhaba [color=#FF0000] %s [/color] !' % kulladi1
        else:
            self.root.ids.mesaj.text='Kullanıcı adı veya Kullanıcı Şifresi Hatalı'
        

olayUyg().run()
```
-------------------------------------------------
kv code:
```
BoxLayout:
	orientation: 'vertical'
	BoxLayout:
		Label:
			text:"Kullanıcı Adını Giriniz"
		TextInput:
			id: kulladi
	BoxLayout:
		Label:
			text:"Kullanıcı Şifresi Giriniz"
		TextInput:
			id: kullsifre
	BoxLayout:
		Button:
			text:"Giriş Yap"
			on_press: app.sifredogrula()
	BoxLayout:
		#Widget:
		Label:
			id:mesaj
			text:"bilgi"

	BoxLayout:
		Label:
			id:mesaj1
	BoxLayout:
		Label:
			id:mesaj2
	BoxLayout:
		Label:
			id:mesaj3
	BoxLayout:
		Label:
			id:mesaj4
	BoxLayout:
		Label:
			id:mesaj5
	BoxLayout:
		Label:
			id:mesaj6

```

**Comment by sccreeper** [comment]
I ran your code and Arch (5.17.1-arch1-1, Python 3.10.4, Kivy 2.1.0) and got the same error. Here is what I changed to fix it. 
TLDR; I moved the machine info code to the `on_start` method and made the `build` method just return a built kv file.
I hope this helps.

**main.py**

```python
# -*- coding: utf-8 -*-

from platform import machine, version, platform, uname, system, processor
from kivy.app import App
from kivy.lang.builder import Builder

makineadi1=machine()
makine_versionu=version()
makine_platform=platform()
makine_adi=uname()
makine_system=system()
makine_islemci=processor()

class olayUyg(App):
    def build(self):
        return Builder.load_file("main.kv")

    def sifredogrula(self):
        kulladi1 = self.root.ids.kulladi.text
        kullsifre1=self.root.ids.kullsifre.text
        if kulladi1=="DexmoD" and kullsifre1=="DexmoD":
            self.root.ids.mesaj.text='Merhaba [color=#FF0000] %s [/color] !' % kulladi1
        else:
            self.root.ids.mesaj.text='Kullanıcı adı veya Kullanıcı Şifresi Hatalı'

    def on_start(self):
        self.root.ids.mesaj1.text='makine info: {} !'.format(makineadi1)
        self.root.ids.mesaj2.text='makine versionu: {} !'.format(makine_versionu)
        self.root.ids.mesaj3.text='makine platform: {} !'.format(makine_platform)
        self.root.ids.mesaj4.text='makine adı: {} !'.format(makine_adi)
        self.root.ids.mesaj5.text='makine system: {} !'.format(makine_system)
        self.root.ids.mesaj6.text='makine işlemci: {} !'.format(makine_islemci)      

olayUyg().run()
```

**main.kv**

```kvlang
BoxLayout:
	orientation: 'vertical'
	BoxLayout:
		Label:
			text:"Kullanıcı Adını Giriniz"
		TextInput:
			id: kulladi
	BoxLayout:
		Label:
			text:"Kullanıcı Şifresi Giriniz"
		TextInput:
			id: kullsifre
	BoxLayout:
		Button:
			text:"Giriş Yap"
			on_press: app.sifredogrula()
	BoxLayout:
		#Widget:
		Label:
			id:mesaj
			text:"bilgi"

	BoxLayout:
		Label:
			id:mesaj1
	BoxLayout:
		Label:
			id:mesaj2
	BoxLayout:
		Label:
			id:mesaj3
	BoxLayout:
		Label:
			id:mesaj4
	BoxLayout:
		Label:
			id:mesaj5
	BoxLayout:
		Label:
			id:mesaj6
```

**Comment by cuneyttanrisever** [comment]
> 
thank you so much.
It worked like this. My problem was that the kv filename had one uppercase letter in it.
